debezium-core/src/main/java/io/debezium/connector/common/BaseSourceTask.java

```diff
@@ -175,16 +172,16 @@ void logStatistics(final List<SourceRecord> records) {
             return;
         }
         int batchSize = records.size();
-        recordCounter += batchSize;
+
         if (batchSize > 0) {
             SourceRecord lastRecord = records.get(batchSize - 1);
             lastOffset = lastRecord.sourceOffset();
             if (pollOutputDelay.hasElapsed()) {
                 // We want to record the status ...
                 final Instant currentTime = clock.currentTime();
-                LOGGER.info("{} records sent during previous {}, last recorded offset: {}", recordCounter,
+                LOGGER.info("{} records sent during previous {}, last recorded offset: {}", batchSize,
                         Strings.duration(Duration.between(previousOutputInstant, currentTime).toMillis()), lastOffset);
-                recordCounter = 0;
```

Comment:
Why is it necessary to drop this one? Can't the log message be change to contain both batch size and the total message count?

Reply:
Sure, it could. But we expose the total count via metrics, so I don't see much value in doing it again via logs?